pkg/providers/instance/filter/filter_test.go

```diff
@@ -508,145 +508,137 @@ var _ = Describe("InstanceFiltersTest", func() {
 		})
 	})
 
-	Context("SpotInstanceFilter", func() {
-		It("should reject spot instances whose cheapest offering is more expensive than the cheapest on-demand offering", func() {
-			f := filter.SpotInstanceFilter(scheduling.NewRequirements(scheduling.NewRequirement(
-				karpv1.CapacityTypeLabelKey,
-				corev1.NodeSelectorOpExists,
-			)))
+	Context("SpotOfferingFilter", func() {
+		It("should filter out expensive spot offerings while keeping instance types with remaining offerings", func() {
+			f := filter.SpotOfferingFilter(scheduling.NewRequirements(
+				scheduling.NewRequirement(
+					karpv1.CapacityTypeLabelKey,
+					corev1.NodeSelectorOpExists,
+				),
+				scheduling.NewRequirement(
+					"test.karpenter.sh/tag",
+					corev1.NodeSelectorOpIn,
+					"kept",
+				),
```

Comment:
We don't want the filter requirements to have any specific tag since that could result in offerings being filtered out due to that incompatibility. Whenever we're using tags we should use `NodeSelectorOpExists` so it doesn't filter any instance type.